packages/client-utils/src/chain.js

```diff
@@ -37,3 +41,56 @@ export const pollBlocks = opts => async lookup => {
     }
   }
 };
+
+/**
+ * @import {CosmosAPI} from './grpc-rest-api.js';
+ * @import {IntervalIO} from './clock-timer.js';
+ */
+/**
+ * @param {CosmosAPI} api
+ * @param {number} [height]
+ */
+export const queryBlock = (api, height) =>
+  api.getJSON(`/cosmos/base/tendermint/v1beta1/blocks/${height || 'latest'}`);
+
+/**
+ * @param {CosmosAPI} api
+ * @param {number} [delta]
+ */
+const recentBlockRate = async (api, delta = 2) => {
+  const relevant = ({ height, time }) => ({ height, time });
+  const { block: latest } = await queryBlock(api);
+  const heightRecent = Number(latest.header.height) - delta;
+  const { block: recent } = await queryBlock(api, heightRecent);
+  const t0 = Date.parse(recent.header.time);
+  const t1 = Date.parse(latest.header.time);
+  return {
+    delta,
+    latest: { ...relevant(latest.header) },
+    recent: { ...relevant(recent.header) },
+    elapsed: t1 - t0,
+    period: (t1 - t0) / delta,
+  };
+};
+
+/**
+ * @param {IntervalIO & { api: CosmosAPI, delta?: number }} io
```

Comment:
please document the features of this function that required calculating the sampling period.